core/themes/bartik/css/colors.css

```diff
@@ -25,6 +25,9 @@ body {
 .l-header ul.menu li.active-trail a {
   color: #fefeff;
 }
+.l-header .form-item-search-block-form input {
+  color: #000;
+}
```

Comment:
If this color is always going to be black, then this should be moved to style.css. However, I think it would make sense to match the body text color, so using `#3b3b3b` would make this match whatever color is configured for text color.

I'd also say we should remove the `.form-item-search-block-form` class and just apply this to all `input` elements in the header.